src/grids.jl

```diff
@@ -0,0 +1,39 @@
+struct Grid1D
+    nx::Int
+    dx::Float64
+    Lx::Float64
+    xu::Vector
+    xt::Vector
+end
+
+function Grid1D(nx, x_beg, x_end)
+    dx = (x_end - x_beg)/(nx + 0.5)
+    xu = LinRange(x_beg, x_end - dx/2, nx)
+    xt = LinRange(x_beg + dx/2, x_end, nx)
+
+    return Grid1D(nx, dx, Lx, xu, xt)
```

Comment:
`Lx` is not known here to be passed to the `Grid1D` constructor as an argument...

I guess you can compute `Lx = x_end - x_beg`. But What I suggest is that instead of `x_beg` and `x_end` we use `Lx` and `x₀`. Then `x_end = x₀ + Lx`.

Reply:
Another commend is that `x_beg` is bad naming... It does not mean anything. Characters are free; we are not restricted by 8-character variable names as was the case in the 1990s. :)

Let's try to use naming convention style [described here](https://docs.julialang.org/en/v1/manual/style-guide/#Use-naming-conventions-consistent-with-Julia-base/)?

Reply:
One more remark here: The `Grid1D` type definition prescribes `xu` of type `Vector` but then the constructor creates `xu` as a range.

```Julia
julia> x = LinRange(0, 2.0, 6)
6-element LinRange{Float64}:
 0.0,0.4,0.8,1.2,1.6,2.0

julia> typeof(x)
LinRange{Float64}

julia> typeof(x) <: Vector
false

julia> typeof(collect(x)) <: Vector
true
```

Reply:
Thanks for these suggestions. I agree with them. Although, wouldn't it be better if we have two Grid functions with these differences:
 1. Option for initial value and length.
 2. Option for x_beginning and x_end.

This will make the code more general.

Reply:
Agree, we can do that via optional keyword arguments in the grid constructor.

Let's do one thing first (pick one, it's the same for me) and we can improve by adding more functionality later. We don't need to make it as general as possible from the beginning.

Reply:
a better notation instead of `x_beginning` and `x_end` would be that the constructor gets a keyword argument like:

```Julia
function Grid1D(nx, x=(0, 1.0))
    xleft, xright = x[1], x[2]
   ....
end